doc/releases/changelog-dev.md

```diff
@@ -42,6 +42,12 @@
   method is not overridden.
   [(#8686)](https://github.com/PennyLaneAI/pennylane/pull/8686)
 
+* Added the `~pennylane.estimator.templates.UnaryIterationQPE` subroutine in the `pennylane.estimator`
+  module. It is a variant of the Quantum Phase Estimation algorithm
+  that uses unary iteration, which can reduce T and Toffoli gate counts at the cost of
+  increasing the number of ancilla qubits required.
```

Comment:
Nit pick, but I think we should try to use auxiliary instead of ancillary where possible. 
```suggestion
  increasing the number of auxiliary qubits required.
```